### PR TITLE
Ds 3152

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

**Comment by JustinCCYap** [comment]
Legend needs to be hidden from export when hidden in Displayr

**Comment by chschan** [comment]
Added code to hide legend if specified in f0d185d

File: R/cchart.R [hunk 1]
```diff
@@ -269,19 +269,273 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
...
+        res$BubbleScale = args$marker.size * 10
+    }
+    return(res)
+}
+
+#' @importFrom grDevices col2rgb rgb2hsv
+autoFontColor <- function (colors)
+{
+    tmp.rgb <- col2rgb(colors)
+    tmp.lum <- apply(tmp.rgb, 2, function(x) return(0.299*x[1] + 0.587*x[2] + 0.114*x[3]))
```
Comment (JustinCCYap) [inline R/cchart.R:533]: Add a comment to explain where these numbers come from

File: R/cchart.R [hunk 2]
```diff
@@ -269,19 +269,273 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
...
+                BackgroundColor = sprintf("%s%X", args$colors[i], round(tmp.opacity*255)))
+
+    # Initialise return output
+    res <- list()
+    if (length(series.settings) > 0)
+        res$TemplateSeries = series.settings
+    if (isTRUE(args$legend.show == FALSE) || isTRUE(args$legend.show == "Hide"))
+        res$ShowLegend <- FALSE
+    res$Legend = list(Font = list(color = args$legend.font.color,
+            family = args$legend.font.family, size = args$legend.font.size/1.3333))
```
Comment (JustinCCYap) [inline R/cchart.R]: Why divide by 1.3333? Perhaps it should be a variable with a descriptive name?

Comment (JustinCCYap) [reply R/cchart.R]: Or a function with a descriptive name that divides by 1.3333

Comment (chschan) [reply R/cchart.R]: Replaced the 1.333 by a function called `px2pt` in a060c6a

File: R/cchart.R (hunk 1 above)
Comment (chschan) [reply R/cchart.R:533]: Added a comment to explain autoFontColor in a060c6a

**Pull request merged**